flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -190,6 +199,11 @@ public static TaskManagerServices fromConfiguration(
 
 		final JobLeaderService jobLeaderService = new JobLeaderService(taskManagerLocation);
 
+		final CheckpointCacheManager checkpointCacheManager = new CheckpointCacheManager(
+			new ScheduledThreadPoolExecutor(1),
+			Executors.directExecutor(),
+			taskManagerServicesConfiguration.getTmpDirPaths()[0]);
```

Comment:
I find this problematic, because it does not consider all the configured tmp directories. While most user probably have only a single tmp directory configured, this can be problematic if somebody makes use of multiple directories (e.g. to utilize multiple smaller disks). We should also be sensitive about this case.

Reply:
You're right. I even thought we should add a configuration for local recovery to store the data.